test/responders/gif_me_test.exs

```diff
@@ -0,0 +1,13 @@
+defmodule Elix.Responders.GifMeTest do
+  use Hedwig.RobotCase
+  use ExVCR.Mock, adapter: ExVCR.Adapter.Hackney
+
+  @tag start_robot: true, name: "elix", responders: [{Elix.Responders.GifMe, []}]
+  test "gif me - responds with a GIF url", %{adapter: adapter, msg: msg} do
+    use_cassette "giphy_search_steve" do
+      send adapter, {:message, %{msg | text: "gif me steve"}}
+      assert_receive {:message, %{text: text}}
+      assert String.ends_with?(text, ".gif")
+    end
+  end
+end
```

Comment:
[`ex_vcr`](https://github.com/parroty/exvcr) (where `use_casette` comes from) is ultimately a mocking library, and José [famously opposes this kind of mock](https://twitter.com/josevalim/status/641617411242913792) in Elixir. See the commit https://github.com/indyelixir/elix/pull/1/commits/623cdc994f27b017a3a06b445b769898371a3143 for an  alternative which:

- extracts the HTTP request-making code into a separate module (`GiphyClient`)
- uses a different module in the test environment, which returns test data rather than make HTTP requests (`GiphyClientMock`—maybe not the right name since this is different than )

José recommends this approach in the Twitter convo above, and I followed a [great post from the Hashrocket folks](https://hashrocket.com/blog/posts/mocking-api-s-with-elixir) in doing this. However, it has some issues:

- if we extract the HTTP request-making logic into its own module, we'd *still* need something like `ex_vcr` to test *that*, right?
- I don't love coupling this behavior to the environment, meaning you have to go look in `prod.exs` to ultimately figure out how `Responders.GifMe` does its job.

Still, who am I to disagree with the creator of Elixir? I'd like to use the `GiphyClientMock` approach instead of `ex_vcr`, but present both here for discussion.